src/ctap/storage.rs

```diff
@@ -303,24 +302,41 @@ impl PersistentStore {
 
     /// Returns the PIN hash if defined.
     pub fn pin_hash(&self) -> Result<Option<[u8; PIN_AUTH_LENGTH]>, Ctap2StatusCode> {
-        let pin_hash = match self.store.find(key::PIN_HASH)? {
+        let pin_properties = match self.store.find(key::PIN_PROPERTIES)? {
             None => return Ok(None),
-            Some(pin_hash) => pin_hash,
+            Some(pin_properties) => pin_properties,
         };
-        if pin_hash.len() != PIN_AUTH_LENGTH {
+        if pin_properties.len() != PIN_AUTH_LENGTH + 1 {
             return Err(Ctap2StatusCode::CTAP2_ERR_VENDOR_INTERNAL_ERROR);
         }
-        Ok(Some(*array_ref![pin_hash, 0, PIN_AUTH_LENGTH]))
+        Ok(Some(*array_ref![pin_properties, 1, PIN_AUTH_LENGTH]))
     }
 
-    /// Sets the PIN hash.
+    /// Returns the length of the currently set PIN if defined.
+    #[cfg(feature = "with_ctap2_1")]
+    pub fn pin_code_point_length(&self) -> Result<Option<u8>, Ctap2StatusCode> {
+        let pin_properties = match self.store.find(key::PIN_PROPERTIES)? {
+            None => return Ok(None),
+            Some(pin_properties) => pin_properties,
+        };
+        if pin_properties.len() != PIN_AUTH_LENGTH + 1 {
+            return Err(Ctap2StatusCode::CTAP2_ERR_VENDOR_INTERNAL_ERROR);
+        }
+        Ok(Some(pin_properties[0]))
+    }
```

Comment:
We can have a private `fn pin_properties(&self) -> Result<PinProperties, Ctap2StatusCode>` to factorize both functions.

```rust
pub struct PinProperties {
    hash: [u8; PIN_AUTH_LENGTH],
    code_point_length: u8,
}
```

Then `set_pin` could also take a `PinProperties` if it's convenient at call-site.

Reply:
`PinProperties` done. Since the getters are version dependent and separate, I decided to also leave the setter as is and not expose `PinProperties`.